contracts/Staking/Vesting.sol

```diff
@@ -61,22 +58,27 @@ contract Vesting is OwnableUpgradeable {
     }
 
     function claim(uint256 from, uint256 to) external {
+        require(from < to, "Vesting: 'to' must be larger than 'from'");
         VestingSchedule[] storage userVestingSchedules = vestingSchedules[msg.sender];
 
         uint256 rewardsToClaim = 0;
-        uint256 userVestingSchedulesCount = userVestingSchedules.length;
-        for (uint256 i = from; i < to && i < userVestingSchedulesCount; i++) {
-            if (isFullyVested(userVestingSchedules[i])) {
-                rewardsToClaim = rewardsToClaim.add(userVestingSchedules[i].totalVestedAmount_d18.sub(userVestingSchedules[i].releasedAmount_d18));
+        uint256 userVestingSchedulesLength = userVestingSchedules.length;
+
+        for (uint256 index = from + 1; index <= to && index <= userVestingSchedulesLength; ++index) {
```

Comment:
That would not work IMO.
If you'll do that, the next iteration of the loop i would be 1 (due to the ++i).
Which means you'll skip the vesting schedule in index 0 (which you did not check before since you moved the elements in the array).
So basically of the first vesting schedule has fully vested already, and also the second one - you'll skip the second one.

This is why I chose this approach here, not to change the code too much but also cover all cases..

WDYT?

P.S
Regarding the name of the index, I personally hate the use of i and find it less readable :)
But I guess it's a NIT as both of us doesn't care about it too much I believe

Reply:
Actually what I just wrote applies if you would have just not increment the index if it's zero (like continue does), but in your example whenever the first schedule is fully vested we'll stop and not claim the rest up until the `to` index, which is not good as well